riscv/src/compiler.rs

```diff
@@ -365,32 +343,6 @@ fn replace_dynamic_label_reference(
     ))
 }
 
-fn remove_matching_and_next<I: Iterator, F>(iter: I, predicate: F) -> impl Iterator<Item = I::Item>
-where
-    F: Fn(&I::Item) -> bool,
-{
-    iter.scan(false, move |filter_next, item| {
-        let mut filter_current = *filter_next;
-        *filter_next = predicate(&item);
-        // if the predicate says this line should be filtered, then
-        // the next one should be filtered as well.
-        filter_current |= *filter_next;
-        Some((filter_current, item))
-    })
-    .filter_map(|(filter, statement)| (!filter).then_some(statement))
-}
-
```

Comment:
Why were these removed?

Reply:
they were not being used anymore, as there is no need for "stub replacement"